aiohttp/client.py

```diff
@@ -667,28 +675,49 @@ async def _request(
                         max_field_size=max_field_size,
                     )
 
-                    try:
+                    # Core request handler - use default parameter to avoid loop variable issues
+                    async def _send_request(
```

Comment:
I think this ends up running too late for the case we want to block the connection

Reply:
adjusted